packages/service-provider-core/src/printable-bson.ts

```diff
@@ -43,10 +44,38 @@ export const bsonStringifiers: Record<
   BSONRegExp: makeClasslessInspect('BSONRegExp'),
   Binary: function (
     this: typeof BSON.Binary.prototype,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
     ...args: any[]
   ): string {
     const hexString = this.toString('hex');
     switch (this.sub_type) {
+      case BSON.Binary.SUBTYPE_VECTOR:
+        switch (this.buffer[0]) {
+          case BSON.Binary.VECTOR_TYPE.Int8:
+            return `Int8(${utilInspect(
+              Array.from(this.toInt8Array()),
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+              ...args
+            )})`;
+          case BSON.Binary.VECTOR_TYPE.Float32:
+            return `Float32(${utilInspect(
+              Array.from(this.toFloat32Array()),
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+              ...args
+            )})`;
+          case BSON.Binary.VECTOR_TYPE.PackedBit:
+            return `PackedBit(${utilInspect(
+              hexString,
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+              ...args
+            )})`;
+          default:
+            return `Vector(${utilInspect(
```

Comment:
In general, the representation we go for in our inspect function is one that allows re-creating the object by pasting the output back into input, unless we have a reason not to do that here, let's keep that intact